cpp/src/arrow/c/bridge.h

```diff
@@ -321,6 +321,31 @@ ARROW_EXPORT
 Status ExportChunkedArray(std::shared_ptr<ChunkedArray> chunked_array,
                           struct ArrowArrayStream* out);
 
+/// \brief Export C++ RecordBatchReader using the C device stream interface
+///
+/// The resulting ArrowDeviceArrayStream struct keeps the record batch reader
+/// alive until its release callback is called by the consumer. The device
+/// type is determined by calling device_type() on the RecordBatchReader.
+///
+/// \param[in] reader RecordBatchReader object to export
+/// \param[out] out C struct to export the stream to
+ARROW_EXPORT
+Status ExportDeviceRecordBatchReader(std::shared_ptr<RecordBatchReader> reader,
```

Comment:
Potentially related to my other comment, but the existing `ExportDeviceArray` has a `sync` keyword that the user of this API needs to provide. Is there a reason those methods don't have that?  
The returned `ArrowDeviceArrayStream` itself doesn't have a sync event member, but the `ArrowDeviceArray`s that it will return still have that. The user shouldn't pass the sync event to set in those arrays?

Reply:
This is why I put the `GetSyncEvent` on the `RecordBatch` object since there could potentially be a different sync event for each record batch returned by the reader. Right now it currently is hardcoded to return null, but the `GetSyncEvent` function is virtual, so someone could potentially have a `RecordBatchReader` that returns a class which inherits from `RecordBatch` that implements `GetSyncEvent` to return the correct event object etc.

Reply:
I've made a bunch of changes propagating the sync event and the device type now, so `GetSyncEvent` now does something :) 

Thoughts?